File: c++/from_generator_example/mem_from_gen_example.cpp
#include <iostream>
#include <vector>
#include "../memory_estimators_with_co_order_coll_head.h"
#include "../Synthetic_generators_head.h"
//Generate and test memory for a single DARN(p) network.

using namespace std;


int main(int argc, char *argv[]) {
	
	int nodes = 20;
	int mem_len = 4;
	double mem_strength = 0.9;
	double avg_prob = 0.1;
	int time_steps = 100000;
	int wait_time = 100;			//Time to wait for the model to reach a steady state.
	int max_order = 10;
	int gen_seed = 12353423;		//For seeding the random number generator.
	
	vector<vector<bool> > edge_series = DARNp_generator(nodes, mem_len, mem_strength, avg_prob, time_steps, wait_time, gen_seed);
	int memory_estimate = get_network_effective_memory(edge_series, max_order, 0);
	
	cout << "The estimated memory is: " << memory_estimate << endl;
	
	return 0;
}

File: python_3_paralell/from_generator/mem_from_gen_example.py
import DARNp_generator
import memory_estimators_paralell

"""
This program serves as an example for how to use the memory estimator with a generator program.
Paralell version.
"""

def main():
	nodes = 3
	mem_len = 4
	mem_strength = 0.9
	avg_prob = 0.1
	time_steps = 100000
	max_order = 10
	
	edge_series = DARNp_generator.generate_DARNp_net(nodes, mem_len, mem_strength, avg_prob, time_steps)
	
	memory_estimate = memory_estimators_paralell.get_network_effective_memory(edge_series, max_order, 0)
	
	print("The estimated memory is: %d" %(memory_estimate))

if __name__ == "__main__":
	main()
